Fix path parameter validation

raymondbutcher opened the pull request:
This change, along with the updates to fleek-parser and fleek-router (thanks for merging them so quickly!), gives us working path parameter validation.

johnhof commented on lib/validate/ctx.js:
```diff
@@ -32,7 +33,7 @@ module.exports = function (ctx, parameters) {
       case 'query' : src = request.query || {}; break;
       case 'body'  : src = request.body || {}; break;
-      case 'path'  : src = request.params || {}; break;
+      case 'path'  : src = requestParams || {}; break;
```
I may be missing something, but I'm not totally sure what this is for. shouldn't it already default?

raymondbutcher replied:
Koa-router sets URL parameters in `ctx.params` and not `ctx.request.params`. See https://github.com/alexmingoia/koa-router#url-parameters

Perhaps I should rename `requestParams` to `urlParameters` or `pathParameters` or something. What do you think?

johnhof commented on lib/validate/ctx.js:
```diff
@@ -20,6 +20,7 @@ var _               = require('lodash');
 //
 module.exports = function (ctx, parameters) {
   var request = (ctx ||{}).request;
```
maybe this should be `var request = (ctx ||{}).request || {};`?

raymondbutcher replied:
I didn't look to see when or if this ever gets called without valid context, but I guess it wouldn't hurt. I'll update it.

johnhof commented:
@raymondbutcher thanks for the contributions! we fell into a bit of a fixed use case, so its hard for us to find errors/improvements

raymondbutcher commented:
@johnhof you're welcome! I figured your use case didn't include URL parameter validation when I tried it out and found it didn't work. But the fleekjs project as a whole is really close to what we want to use for a new API project, so thanks!

raymondbutcher commented:
@johnhof I've updated the branch with these key changes:
* implemented your suggestion of `var request = (ctx ||{}).request || {};`
* removed my extra variable, instead accessing `ctx` directly in the `switch`
* added tests
* **TEMPORARILY** depend on `git://github.com/fleekjs/fleek-parser.git` because the latest version on NPM doesn't have the path fix, somehow

johnhof commented:
@raymondbutcher  sorry about that, my local wasnt up to date when I published. I'll update package.json after the merge before I publish. Thanks again!

raymondbutcher commented:
That's quite alright. Cheers!
